main/githooks.py

```diff
@@ -255,32 +318,97 @@ def trim_trailing_whitespace_in_file(filename, new_file=False):
     else:
         line_nums = get_changed_lines(filename)
 
-    modified_file = False
+    modified_file = False   # if trimming white space in place
+    modified_lines = []     # if flagging instead of trimming in place
 
     for line_num in line_nums:
         try:
             before = lines[line_num-1]
         except IndexError as exc:
-            print(f'Error {exc}: {line_num-1} in {filename}')
+            print(f'Error {exc}: {line_num} in {filename}')
             continue
         after = trim_trailing_whitespace(before)
         if before != after:
-            print(f'   Fixed line {line_num}')
-            modified_file = True
-            lines[line_num-1] = after
+            if in_place:
+                print(f'   Fixed line {line_num}')
+                modified_file = True
+                lines[line_num-1] = after
+            else:
+                modified_lines.append(str(line_num))
 
     if modified_file:
         with open(filename, 'wb') as fileobj:
             lines = ''.join(lines)
             fileobj.write(lines.encode())
         add_file_to_index(filename)
 
+    if modified_lines:
+        _fail(f'Found trailing white space in {filename} at lines: ' +
+              ','.join(modified_lines))
+        return 1
+
+    return 0
+
 
-def remove_trailing_white_space(files, new_files=False):
+def remove_trailing_white_space(files, new_files=False, in_place=True):
     '''Remove trailing white spaces in all new and modified lines'''
+    retval = 0
     for filename in files:
-        print(f'  Checking file {filename}')
-        trim_trailing_whitespace_in_file(filename, new_files)
+        retval += trim_trailing_whitespace_in_file(filename, new_files,
+                                                   in_place)
+    return retval
+
+
+def check_filename(filepath):
```

Comment:
Similarly it would be good to see some tests of these file patterns

Reply:
Tests added.
Please see https://github.com/ccdc-opensource/commit-hooks/pull/6